src/protagonist.h

```diff
@@ -9,6 +9,13 @@
 #include "SectionParserData.h"
 #include "SourceAnnotation.h"
 
+struct optionsResult {
```

Comment:
Please rename the `optionsResult` struct as `OptionsResult`. And if possible, move this and the `ParseOptionsObject` function to be under `protagonist` namespace below.